crates/bevy_reflect/src/serde/de.rs

```diff
@@ -1421,7 +1426,9 @@ mod tests {
             .deserialize(&mut ron_deserializer)
             .unwrap();
 
-        let output = <Foo as FromReflect>::from_reflect(dynamic_output.as_ref()).unwrap();
+        let output =
+            <Foo as FromReflect>::from_reflect(dynamic_output.as_ref().as_partial_reflect())
```

Comment:
Is the `as_ref` necessary still? Or, if it is, is `as_partial_reflect` necessary?

I ask because shouldn't `dynamic_output` already be a `Box<dyn PartialReflect>`?